### PR TITLE
Using Git and GitHub on the Windows platform

### PR DESCRIPTION
**Pull request by beardofedu**
## Overview
**TL;DR**
Create course for how to git and github from the Windows platform.

### Questions
I'm sure I am going to have questions, but the biggest one I have currently is:

- If you were moving from macOS to Windows, what would you want to know about the git experience?
- If you were coming from a different version control system, what would you want to know about?

### Next Steps
- [ ] Flesh out the rest of the outline
- [ ] More content about TFS integration
- [ ] Identify tell me why content that needs to be reused and potentially re-chunked

### Review

No pressing need for 👀  yet. Will call for review in a later comment.

### CODE REVIEW

**Comment by beardofedu** [comment]
@brianamarie / @crichID / @saraford 

So, the content is in a general "rough draft" state. I need some 👀  to review the course and provide some 💭 s on what content I need to include, I am including a checklist / review sheet that should be easy to use when looking at the content.

Additionally, @saraford if there is additional content you need added to the course, please let me know. I am still planning on having this available in a `beta` form by the time you have your meeting next week.

```
# Welcome to GitHub for Windows Users
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Get Started
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Install Git
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Configure Git
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# CRLF aka Line Endings
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Store Your Password
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Editors
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# vi Basics
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Configure Other Editors
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Visual Source Integration
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Install the GitHub Extension
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Start a Project
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Make a Commit
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

# Sync?
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

## Content review

```

**Comment by brianamarie** [comment]
General feedback:
- How do we get to this page? Will we add to the navbar/timeline, or for this class will it be URL specific?
- Continue buttons aren't working properly :(
- I went through this once on Mac looking for formatting things, and once on Windows going through the actual steps. Notes about roadbumps/places we could add clarity are included. 😄 
- This is SO RAD and it's very impressive that you've hammered out so much content so quickly. :heart: I dream of a future where we can automagically detect the operating system of a user and integrate this content into our existing offerings for all project types. :sparkles: Thank you.

### Welcome to GitHub for Windows Users
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why
- Maybe not start both sentences with "although"

### Get Started
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why
- Do we need this page to be separate from "Install Git"? What if we had this header and the content for Install Git on this same page?

### Install Git
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why

### Configure Git
- [x] Needs Show Me How
- [ ] Needs Tell Me Why
- s/your_email.mail.com/your_email@email.com

### CRLF aka Line Endings
- [x] Needs Show Me How
- [x] Needs Tell Me Why
- Title including "aka" might lose some with English as a second language

### Store Your Password
- [X] Needs Show Me How
- [ ] Needs Tell Me Why **note: There appears to be info here, but the includes may not be set up correctly**
- This may have happened automagically last time I pushed, because I remember doing something vaguely like that when prompted, and nothing prompted me this time.

### Editors
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why
- Same feedback as the "Get Started" page - do we need this extra page? Does it add value?

### vi Basics
- [ ] Needs Show Me How **note: It looks like there's an image supposed to be here, but the includes isn't working properly**
- [ ] Needs Tell Me Why **note: the current tell me why is for configs**
- I don't know if it's supposed to be "VI" or "vi" or if it doesn't matter, but it's displayed both ways within this page. /shrug

### Configure Other Editors
- [ ] Needs Show Me How **note: It looks like there's an image supposed to be here, but the includes isn't working properly**
- [ ] Needs Tell Me Why **note: the current tell me why is for configs**
- This looks awesome
- This didn't work for me, I had to add atom to my PATH which was a pain and I imagine it's different for different versions of Windows. Is there windows documentation we can point to that we can trust?

### Visual Source Integration
- [ ] Needs Show Me How
- [ ] Needs Tell Me Why
- This URL gets me "not found": /on-demand/windows/visual-source

### Install the GitHub Extension
- [ ] Needs Show Me How **note: It looks like there's an image supposed to be here, but the includes isn't working properly**
- [ ] Needs Tell Me Why
- Spelling in first sentence - isn't'tuncommon/isn't uncommon, also that sentence may be a run on sentence, sort of like this sentence.
- The instructions there are slick and look great

### Start a Project
- [ ] Needs Show Me How **note: It looks like there's an image supposed to be here, but the includes isn't working properly**
- [ ] Needs Tell Me Why
- :sparkles: :heart:
- Sometimes CLICK is in all caps
- Do we want to include instructions for how to import GitHub repositories?


Alternate instructions for starting locally:
- Open the GitHub integration window: View > Other Windows > GitHub
- You may be able to push to GitHub, or you may be told there is no source control. Depending on what you see here, you may need to click Project > Add to Source control
- In the GitHub window, push the green button to publish your code "Get Started"
- Under "Publish to GitHub", click "Publish"
- Make sure the name and description matches your project, and select if you'd like the repository to be private.


### Make a Commit
- [x] Needs Show Me How
- [ ] Needs Tell Me Why
- This is awesome!

I think we could get ride of the first section, and only have "Committing all of the changes" and "Committing some of the changes". Also, I think we should consider adding a step about going to "Project" and "changes". I didn't see that for a while.

For committing all of the changes:
- In the Team Explorer pane under "Project", click "Changes".
- Enter a commit message and make sure the branch listed is the correct branch.

For committing some of the changes:
- In the Team Explorer pane under "Project", click "Changes".
- Right click the file that you'd like to be committed and select "stage".
- Enter a commit message and make sure the branch listed is the correct branch. **The commit will only include the changes in the files that are under "Staged Changes".**

### Sync?
- [ ] Needs Show Me How **note: It looks like there's an image supposed to be here, but the includes isn't working properly**
- [x] Needs Tell Me Why
- I think you intend for there to be a few bulleted lists, but it isn't coming across
- Is there a reason there's a question mark in the title?

**Comment by beardofedu** [comment]
Things to add based on convo with Windows peeps:

1. Example of branching, show on whiteboard, GH desktop, then within VS
2. Why git
3. setup walkthrough
4. git shell demos (show that it exists, and you don't have to login)
5. merge conflicts
6. demo beyondcompare / merge conflict resolution tools

**Comment by beardofedu** [comment]
@brianamarie I know this is _kinda_ rushed, but can you check this Monday morning and if everything seems _decent_ enough to publish as a **hard** beta click the Big Green and merge `services-web` so we can let @saraford point to it during her talk tomorrow?

**Comment by beardofedu** [comment]
cc: @crichID I _think_ @brianamarie might be ✈️ Monday morning ... so she might not be able to review this prior to tomorrow's class. Your 👀 would be super appreciated.

**Comment by brianamarie** [comment]
@beardofedu 👍 from me. I made some commits, most of them have to do with the navigation and the 'continue' buttons, and one has to do with rewording. I think that we can still fine tune the instructions for adding a commit in VS, but it might be best to see what feedback @saraford gets from the current state with real life users. 

So many ✨, @beardofedu!!!

**Comment by beardofedu** [comment]
Merging this. Course is in a `beta` format and only available from the drop-down menu.
